pkg/reconciler/subscription/subscription.go

```diff
@@ -271,36 +271,35 @@ func (r *Reconciler) resolveDeadLetterSink(ctx context.Context, subscription *v1
 	if subscription.Spec.Delivery != nil && subscription.Spec.Delivery.DeadLetterSink != nil {
 		deadLetterSinkAddr, err := r.destinationResolver.AddressableFromDestinationV1(ctx, *subscription.Spec.Delivery.DeadLetterSink, subscription)
 		if err != nil {
-			subscription.Status.PhysicalSubscription.DeadLetterSinkURI = nil
+			subscription.Status.PhysicalSubscription.DeliveryStatus = eventingduckv1.DeliveryStatus{}
 			logging.FromContext(ctx).Warnw("Failed to resolve spec.delivery.deadLetterSink",
 				zap.Error(err),
 				zap.Any("delivery.deadLetterSink", subscription.Spec.Delivery.DeadLetterSink))
 			subscription.Status.MarkReferencesNotResolved(deadLetterSinkResolveFailed, "Failed to resolve spec.delivery.deadLetterSink: %v", err)
 			return pkgreconciler.NewEvent(corev1.EventTypeWarning, deadLetterSinkResolveFailed, "Failed to resolve spec.delivery.deadLetterSink: %w", err)
 		}
-		deadLetterSinkURI := deadLetterSinkAddr.URL
 
-		logging.FromContext(ctx).Debugw("Resolved deadLetterSink", zap.String("deadLetterSinkURI", deadLetterSinkURI.String()))
-		subscription.Status.PhysicalSubscription.DeadLetterSinkURI = deadLetterSinkURI
+		logging.FromContext(ctx).Debugw("Resolved deadLetterSink", zap.String("deadLetterSinkURI", deadLetterSinkAddr.URL.String()))
```

Comment:
nit: WDYT about logging the whole DLS object (including the ca certs)? Just to have this information and it's on debug level anyhow...